runtime/common/src/traits.rs

```diff
@@ -137,6 +137,47 @@ pub trait Leaser {
 	fn lease_period_index() -> Self::LeasePeriod;
 }
 
+/// An enum which tracks the status of the auction system, and which phase it is in.
+#[derive(PartialEq, Debug)]
+pub enum AuctionStatus<BlockNumber> {
+	/// An auction has not started yet.
+	NotStarted,
+	/// We are in the opening period of the auction, collecting initial bids.
+	OpeningPeriod,
+	/// We are in the ending period of the auction, where we are taking snapshots of the winning
+	/// bids. This state supports "sampling", where we may only take a snapshot every N blocks.
+	/// In this case, the first number is the current sample number, and the second number
+	/// is the sub-sample. i.e. for sampling every 20 blocks, the 25th block in the ending period
+	/// will be `EndingPeriod(1, 5)`.
+	EndingPeriod(BlockNumber, BlockNumber),
```

Comment:
nit: Just for symetry sake you could stick to opening/closing or starting/ending

Reply:
yup, agreed